File: Java_Programs_for_beginners/selectionSort.java
/* Selection Sort */

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Solution {

	public void selectionSort(int arr[], int n) {
		for (int i = 0; i < n - 1; i++) {
			int idx = i;
			for (int j = i + 1; j < n; j++) {
				if (arr[i] > arr[j]) {
					idx = j;
				}
			}
			int temp = arr[i];
			arr[i] = arr[idx];
			arr[idx] = temp;
		}
		for (int i = 0; i < n; i++) {
			System.out.print(arr[i] + " ");
		}
	}

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int num = Integer.parseInt(br.readLine());
		int[] arr = new int[num];
		for (int i = 0; i < num; i++) {
			arr[i] = Integer.parseInt(br.readLine());
		}
		Solution obj = new Solution();
		obj.selectionSort(arr, num);
	}
}

File: selection_sort.py
roll_no = [23,43,41,26,45,53,2,12,74]
print("Selection Sort")
for i in range(len(roll_no)):
    min_val_index=i
    for j in range(i+1,len(roll_no)):
        if roll_no[min_val_index]>roll_no[j]:
            min_val_index=j
    
    roll_no[i],roll_no[min_val_index]=roll_no[min_val_index],roll_no[i]

print(roll_no)